source/services/dataContracts/baseResourceBuilder/baseResourceBuilder.service.ts

```diff
@@ -169,18 +185,78 @@ export class BaseResourceBuilder implements IBaseResourceBuilder {
 	createParentResource<TDataType extends IBaseDomainObject, TSearchParams, TResourceDictionaryType>
 		(options: IParentResourceParams<TDataType, TResourceDictionaryType>): IBaseParentDataService<TDataType, TSearchParams, TResourceDictionaryType> {
 		options.useMock = options.endpoint == null ? true : options.useMock;
-		return new BaseParentDataService(this.$http, this.$q, this.array, options.endpoint, options.mockData, options.resourceDictionaryBuilder, options.transform, options.useMock, options.logRequests);
+		let parentDataService: IBaseParentDataService<TDataType, TSearchParams, TResourceDictionaryType>
+			= new BaseParentDataService(this.$http, this.$q, this.array, options.endpoint, options.mockData, options.resourceDictionaryBuilder, options.transform, options.useMock, options.logRequests);
+		(<any>parentDataService).clone = (endpoint: string): IBaseParentDataService<TDataType, TSearchParams, TResourceDictionaryType> => { return this.cloneParentResource(parentDataService, endpoint); };
+		return parentDataService;
+	}
+
+	createParentResourceView<TDataType extends IBaseDomainObject, TSearchParams, TResourceDictionaryType>
+		(options: IParentResourceParams<TDataType, TResourceDictionaryType>): IBaseParentDataServiceView<TDataType, TSearchParams, TResourceDictionaryType> {
+		let dataServiceView: IBaseParentDataServiceView<TDataType, TSearchParams, TResourceDictionaryType> = <any>this.createParentResource(options);
+		(<any>dataServiceView).clone = (endpoint: string): IBaseParentDataServiceView<TDataType, TSearchParams, TResourceDictionaryType> => { return <any>this.cloneParentResource(dataServiceView, endpoint); };
```

Comment:
Here and below, the only reason cloning a parent and creating a parent resource view would be required is if you have a parent that is the child of another parent. An example of this is performedBy, which is a child of the service event, but has vendor sub-contract and time entry resources under it.